spyder/utils/ipython/spyder_kernel.py

```diff
@@ -212,11 +214,12 @@ def publish_pdb_state(self):
         Publish Variable Explorer state and Pdb step through
         publish_data.
         """
-        if self._pdb_obj:
+        if self._pdb_obj and not self._do_publish_data:
```

Comment:
Let's change the logic here because it's a bit confuse. This should be

```python
if self._pdb_obj and self._do_publish_data:
```

to be clearer.